cmd/zc_traverser_local.go

```diff
@@ -247,7 +247,28 @@ func WalkWithSymlinks(appCtx context.Context, fullPath string, walkFunc filepath
 			}
 
 			if fileInfo.Mode()&os.ModeSymlink != 0 {
-				if !followSymlinks {
+				if symlinkHandling.Preserve() {
+					// Handle it like it's not a symlink
+					result, err := filepath.Abs(filePath)
+
+					if err != nil {
+						WarnStdoutAndScanningLog(fmt.Sprintf("Failed to get absolute path of %s: %s", filePath, err))
+						return nil
+					}
+
+					err = walkFunc(common.GenerateFullPath(fullPath, computedRelativePath), fileInfo, fileError)
+					// Since this doesn't directly manipulate the error, and only checks for a specific error, it's OK to use in a generic function.
+					skipped, err := getProcessingError(err)
+
+					// If the file was skipped, don't record it.
+					if !skipped {
+						seenPaths.Record(common.ToExtendedPath(result))
+					}
+
+					return err
+				}
+
```

Comment:
Instead of duplicating the code (and I see that it does not have the HasSeen() handling), why not simply change L249 to

if  (fileInfo.Mode()&os.ModeSymlink != 0) && !symlinkHandling.Follow() {

then the skip check is the only check we need to do

Reply:
follow symlinks still needs to be handled